services/core-web/src/components/Forms/permits/conditions/VariableConditionMenu.js

```diff
@@ -23,9 +26,19 @@ const defaultProps = {
 
 export class VariableConditionMenu extends Component {
   handleClick(value) {
-    const condition = this.props.formValues.condition ? this.props.formValues.condition : "";
-    const newValues = `${condition} ${value.key}`;
-    return this.props.change(FORM.CONDITION_SECTION, "condition", newValues);
+    const condition = this.props.conditionFormValues.condition
+      ? this.props.conditionFormValues.condition
+      : "";
+    if (condition !== "") {
+      const newValues = `${condition} ${value.key}`;
+      return this.props.change(FORM.CONDITION_SECTION, "condition", newValues);
+    }
+
+    const preambleText = this.props.generatePermitFormValues.preamble_text
+      ? this.props.generatePermitFormValues.preamble_text
+      : "";
+    const newPreambleText = `${preambleText} ${value.key}`;
+    return this.props.change(FORM.GENERATE_PERMIT, "preamble_text", newPreambleText);
   }
 
```

Comment:
I think we can simplify this if/else 

```
if (!isEmpty(this.props.conditionFormValues)) {
// condition logic
} else {
// preamble_text logic
}
```

OR you can pass the form, and the field string into the component and make this function generic.